test/extensions/filters/http/json_to_metadata/integration_test.cc

```diff
@@ -0,0 +1,157 @@
+    if (request_body.empty()) {
+      response = codec_client_->makeHeaderOnlyRequest(request_headers);
+    } else {
+      auto encoder_decoder = codec_client_->startRequest(request_headers);
+      request_encoder_ = &encoder_decoder.first;
+      response = std::move(encoder_decoder.second);
+      size_t i = 0;
+      for (; i < request_body.length() / chunk_size; i++) {
+        Buffer::OwnedImpl buffer(request_body.substr(i * chunk_size, chunk_size));
+        codec_client_->sendData(*request_encoder_, buffer, false);
+      }
+      // Send the last chunk flagged as end_stream.
+      Buffer::OwnedImpl buffer(
+          request_body.substr(i * chunk_size, request_body.length() % chunk_size));
+      codec_client_->sendData(*request_encoder_, buffer, true);
+    }
+    ASSERT_TRUE(fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_));
+    ASSERT_TRUE(fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_));
+    ASSERT_TRUE(upstream_request_->waitForEndStream(*dispatcher_));
+
+    upstream_request_->encodeHeaders(default_response_headers_, true);
+    ASSERT_TRUE(response->waitForEndStream());
+    EXPECT_TRUE(response->complete());
+
+    // cleanup
+    codec_client_->close();
+    ASSERT_TRUE(fake_upstream_connection_->close());
+    ASSERT_TRUE(fake_upstream_connection_->waitForDisconnect());
+  }
+
+  const std::string filter_config_ = R"EOF(
+name: envoy.filters.http.json_to_metadata
+typed_config:
+  "@type": type.googleapis.com/envoy.extensions.filters.http.json_to_metadata.v3.JsonToMetadata
+  request_rules:
+    rules:
+    - selectors:
+      - key: version
+      on_present:
+        metadata_namespace: envoy.lb
+        key: version
+      on_missing:
+        metadata_namespace: envoy.lb
+        key: version
+        value: 'unknown'
+        preserve_existing_metadata_value: true
+      on_error:
+        metadata_namespace: envoy.lb
+        key: version
+        value: 'error'
+        preserve_existing_metadata_value: true
+)EOF";
+
+  Http::TestRequestHeaderMapImpl incoming_headers_{{":scheme", "http"},
+                                                   {":path", "/ping"},
+                                                   {":method", "GET"},
+                                                   {":authority", "host"},
+                                                   {"Content-Type", "application/json"}};
+
+  std::string request_body_ =
+      R"delimiter(
+        {"version":"1.0.0",
+        "messages":[
+          {"role":"user","content":"content A"},
+          {"role":"assistant","content":"content B"},
+          {"role":"user","content":"content C"},
+          {"role":"assistant","content":"content D"},
+          {"role":"user","content":"content E"}],
+        "stream":true})delimiter";
+};
+
+// TODO(#26236): Fix test suite for HTTP/3.
+INSTANTIATE_TEST_SUITE_P(
+    Protocols, JsonToMetadataIntegrationTest,
```

Comment:
I had hard time to understand what to test watermark in json to metadata filter.
I assume watermark is buffer limit here but this filter doesn't really do the local reply based on filter buffer limit and so really i'm testing the test framework in my latest pr instead of testing the json to metadata filter.

I don't have the best knowledge on the filter buffer limit. Therefore I need more help if the buffer limit integration test  is necessary.

Reply:
ack, it might no longer be necessary since we removed the buffering limit this filter has; still the test is failing -- while we don't __need__ it , can you figure out why? We might be missing something.

Reply:
reply in another thread. thanks!